tests/unit/components/Match/PhotosSection.test.js

```diff
@@ -0,0 +1,114 @@
+import { fireEvent, screen, waitFor } from "@testing-library/react-native";
+import PhotosSection from "components/Match/PhotosSection";
+import React from "react";
+import factory from "tests/factory";
+import { renderComponent } from "tests/helpers/render";
+
+describe( "PhotosSection", () => {
+  const mockNavToTaxonDetails = jest.fn();
+
+  const defaultProps = {
+    representativePhoto: factory( "RemotePhoto" ),
+    obsPhotos: [factory( "LocalObservationPhoto" )],
+    navToTaxonDetails: mockNavToTaxonDetails,
+    taxon: factory( "LocalTaxon", {
+      taxonPhotos: [
+        { photo: factory( "RemotePhoto", { id: 1 } ) },
+        { photo: factory( "RemotePhoto", { id: 2 } ) },
+        { photo: factory( "RemotePhoto", { id: 3 } ) }
+      ]
+    } )
+  };
+
+  it( "displays photo count when multiple observation photos exist", async () => {
+    const multipleObsPhotos = [
+      factory( "LocalObservationPhoto" ),
+      factory( "LocalObservationPhoto" ),
+      factory( "LocalObservationPhoto" )
+    ];
+
+    renderComponent(
+      <PhotosSection
+        {...defaultProps}
+        obsPhotos={multipleObsPhotos}
+      />
+    );
+
+    await waitFor( () => {
+      expect( screen.getByTestId( "MatchScreen.ObsPhoto" ) ).toBeTruthy();
+    } );
+    expect( screen.getByText( "3" ) ).toBeTruthy();
+  } );
+
+  it( "hides taxon photos when hideTaxonPhotos is true", async () => {
+    renderComponent(
+      <PhotosSection
+        {...defaultProps}
+        hideTaxonPhotos
+      />
+    );
+
+    await waitFor( () => {
+      expect( screen.queryByTestId( "TaxonDetails.photo.1" ) ).toBeFalsy();
+    } );
+  } );
+
+  // Not working due to known bug: https://linear.app/inaturalist/issue/MOB-1069/taxon-photos-hidden-for-iconic-taxa-match-screens
```

Comment:
I have not looked into this but I am fine to leave it in here, But please add a note to the ticket to enable this test there as well.